src/App.tsx

```diff
@@ -15,71 +17,101 @@ export const goods = [
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+type State = {
+  selectedValue: string,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedValue: 'Jam',
+  };
+
+  clearValue = () => {
+    this.setState({ selectedValue: '' });
+
+    document.querySelectorAll('tr').forEach(el => {
+      el.classList.remove('has-background-success-light');
+
+      const button = el.querySelector('button');
+
+      if (button) {
+        button.innerText = '+';
+        button.classList.remove('is-info');
+        button.dataset.cy = 'AddButton';
+      }
+    });
+  };
+
+  selectValue = (currentTarget: HTMLButtonElement, value: string) => {
+    if (currentTarget.dataset.cy === 'AddButton') {
+      this.clearValue();
+
+      const target = currentTarget;
+      const row = target.closest('tr');
+
+      target.innerHTML = '-';
+      target.classList.add('is-info');
+
+      if (row) {
+        row.classList.toggle(
+          'has-background-success-light',
+        );
+      }
+
+      target.dataset.cy = 'RemoveButton';
+      this.setState({ selectedValue: value });
+    } else {
+      this.clearValue();
+    }
+  };
+
+  render() {
+    return (
+      <main className="section container">
+        {this.state.selectedValue
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              {`${this.state.selectedValue} is selected`}
+
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={this.clearValue}
+              />
+            </h1>
+          )
+          : <h1 className="title">No goods selected</h1>}
+
+        <table className="table">
+          <tbody>
+            {goods.map(item => (
+              <tr
+                data-cy="Good"
+              >
```

Comment:
Wrap if you have > 2 params || length > 80
```suggestion
              <tr data-cy="Good">
```

Reply:
Also, when you make a map - add key to the parent tag
```suggestion
              <tr key={item} data-cy="Good">
```